pkg/githubclient/client.go

```diff
@@ -2,46 +2,171 @@ package githubclient
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"time"
 
 	"github.com/giantswarm/microerror"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/google/go-github/v70/github"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/oauth2"
 )
 
 type Config struct {
-	DryRun bool
-	Logger *logrus.Logger
-
+	Logger      *logrus.Logger
 	AccessToken string
 }
 
 type Client struct {
-	dryRun bool
-	logger *logrus.Logger
-
+	logger      *logrus.Logger
 	accessToken string
+	workDir     string
 }
 
 func New(config Config) (*Client, error) {
 	if config.Logger == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
-
 	if config.AccessToken == "" {
 		return nil, microerror.Maskf(invalidConfigError, "%T.AccessToken must not be empty", config)
 	}
 
 	c := &Client{
-		dryRun: config.DryRun,
```

Comment:
What's it with the `dry-run` feature? It doesn't make sense anymore (understandable, as operations are complex and depend on each other)?